packages/web-extension/src/walletManager/walletManager.types.ts

```diff
@@ -59,17 +73,19 @@ export interface WalletManagerApi {
   /**
    * Deactivates the active wallet and destroy its existing store,
    * so that a future activation of the same wallet creates a new store.
+   *
+   * @param walletId The walletId of the wallet to destroy.
    */
-  destroy(): Promise<void>;
+  destroyData(walletId: WalletId): Promise<void>;
 }
 
 export interface WalletFactory {
   create: (
-    props: WalletManagerActivateProps,
-    dependencies: { keyAgent: AsyncKeyAgent; stores: storage.WalletStores }
+    name: string,
```

Comment:
Since wallet factory is provided by Lace, it needs to know the provider and chain id. I suggest to pass through all the activate props.
```suggestion
    props: WalletManagerActivateProps & { name: string },
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1012/commits/39951537f0ec93792820e128dfc09540d2270a1c